packages/next/src/server/base-server.ts

```diff
@@ -2138,32 +2140,20 @@ export default abstract class Server<ServerOptions extends Options = Options> {
 
     const doRender: Renderer = async (postponed) => {
       // In development, we always want to generate dynamic HTML.
-      const supportsDynamicHTML =
+      const supportsDynamicHTML: boolean =
         // If this isn't a data request and we're not in development, then we
         // support dynamic HTML.
-        (!isDataReq && opts.dev) ||
+        (!isDataReq && opts.dev === true) ||
         // If this is not SSG or does not have static paths, then it supports
         // dynamic HTML.
-        !(isSSG || hasStaticPaths) ||
+        (!isSSG && !hasStaticPaths) ||
         // If this request has provided postponed data, it supports dynamic
         // HTML.
         !!postponed ||
         // If this is a dynamic RSC request, then this render supports dynamic
         // HTML (it's dynamic).
         isDynamicRSCRequest
 
-      // Ensure we bail if a prefetch RSC request is made while PPR is
-      // enabled that has dynamic HTML enabled.
-      if (
-        this.renderOpts.experimental.ppr &&
-        isPrefetchRSCRequest &&
-        supportsDynamicHTML
-      ) {
-        throw new Error(
-          "Invariant: prefetch RSC requests can't support dynamic HTML"
-        )
-      }
```

Comment:
This invariant was being triggered in conditions where it was still allowed, like in `edge`.